src/client/general/intro.md

```diff
@@ -1,23 +1,31 @@
 # General principles
 
-The Wayland protocol follows a client-server architecture: the Wayland compositor listens
-on an UNIX socket, to which client apps then connect in order to present their graphical
+The Wayland protocol follows a client-server architecture: the Wayland compositor[^compositor]
+listens on an UNIX socket, to which client apps then connect in order to present their graphical
 interface. This UNIX connection transports information in both directions: the client uses
-it to tell the server what it wants to display, and the server uses it to tell the client
-about user actions (such as keyboard or pointer input). The Wayland connection is thus the
-heart of a client app, and is the source of events that drives its event loop.
+it to tell the server what it wants to display, and the server tells the client about user
+actions (such as keyboard or pointer input). 
+
+[^compositor]: The terms "compositor" and "server" are used interchangeably in the Wayland ecosystem.
+
+The first step a client will perform is finding Wayland socket and try to connect to it.
 
 The server tells to clients about the Wayland socket using the `WAYLAND_DISPLAY` environment
 variable. The listening socket is placed at `$XDG_RUNTIME_DIR/$WAYLAND_DISPLAY`. This will
 typically be a path like `/run/user/1000/wayland-0`.
 
-When starting your app, it will need to find the Wayland socket and connect to it. The
-wayland-client crate does this for you with the
-[`Display::connect_to_env()`](https://docs.rs/wayland-client/*/wayland_client/struct.Display.html#method.connect_to_env)
-method. If no error occurs, this function will provide you with a `Display` object.
+The server could also tell the client to connect using a specific socket the server has created
+using the `WAYLAND_SOCKET` environment variable. In this case the `WAYLAND_SOCKET` environment
+variable should be interpreted as a file descriptor number. This is a less common way the client
+may connected to the server, but it is sometimes used for specific reasons that will be detailed
+later on in the server section of the book.
```

Comment:
I'm wondering how necessary it is to expose this right from the start, given it's still a pretty nice (while important) case. :thinking:

Reply:
So probably better to mention this in the server section of the book and remove that section here?

Reply:
Sounds good to me, mentioning there are other ways but that they are not yet relevant. Once the parts that talk about it are written we can add a link to them for example.